plugins/woocommerce/client/admin/client/customize-store/index.tsx

```diff
@@ -1,617 +1,278 @@
-) => {
-	const { path } = getQuery() as { path: string };
-	const step = ( action as { step: string } ).step;
-	const pathFragments = path.split( '/' ); // [0] '', [1] 'customize-store', [2] step slug [3] design-with-ai, assembler-hub path fragments
-	if ( pathFragments[ 1 ] === 'customize-store' ) {
-		if ( pathFragments[ 2 ] !== step ) {
-			// this state machine is only concerned with [2], so we ignore changes to [3]
-			// [1] is handled by router at root of wc-admin
-			const newPath = `/customize-store/${ step }`;
-
-			// Since CYS also runs inside an iframe and because the getHistory
-			// creates an instance per window context, making a push to the
-			// history only alters that instance. Here we need to alter the
-			// browser's (window.top) history and not the history of the iframe.
-			if ( isIframe( window ) && window.top ) {
-				// window.location.href does not fit in this case since it produces
-				// a hard refresh to the page.
-				window.top.history.pushState(
-					{},
-					'',
-					getNewPath( {}, newPath )
-				);
-				return;
-			}
 
-			updateQueryString( {}, newPath );
-		}
-	}
-};
-
-const redirectToWooHome = () => {
-	const url = getNewPath( getPersistedQuery(), '/', {} );
-	navigateOrParent( window, url );
-};
+const CustomizeStoreController = () => {
+	useFullScreen( [ 'woocommerce-customize-store' ] );
 
-const goBack = () => {
-	const history = getHistory();
-	if (
-		history.__experimentalLocationStack.length >= 2 &&
-		! history.__experimentalLocationStack[
-			history.__experimentalLocationStack.length - 2
-		].search.includes( 'customize-store' )
-	) {
-		// If the previous location is not a customize-store step, go back in history.
-		history.back();
-		return;
-	}
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
 
-	redirectToWooHome();
-};
+	const currentTheme = useSelect( ( select ) => {
+		// @ts-ignore
+		return select( 'core' ).getCurrentTheme();
+	}, [] );
 
-const markTaskComplete = async () => {
-	const currentTemplateId: string | undefined = await resolveSelect(
-		coreStore
-	).getDefaultTemplateId( { slug: 'home' } );
-	return dispatch( optionsStore ).updateOptions( {
-		woocommerce_admin_customize_store_completed: 'yes',
-		// We use this on the intro page to determine if this same theme was used in the last customization.
-		woocommerce_admin_customize_store_completed_theme_id: currentTemplateId,
-	} );
-};
+	const isBlockTheme = currentTheme?.is_block_theme;
 
-const browserPopstateHandler =
-	() => ( sendBack: Sender< { type: 'EXTERNAL_URL_UPDATE' } > ) => {
-		const popstateHandler = () => {
-			sendBack( { type: 'EXTERNAL_URL_UPDATE' } );
-		};
-		window.addEventListener( 'popstate', popstateHandler );
-		return () => {
-			window.removeEventListener( 'popstate', popstateHandler );
-		};
+	const getDesignUrl = () => {
+		return isBlockTheme
+			? getAdminLink( 'site-editor.php' )
+			: getAdminLink( 'customize.php?return=/wp-admin/themes.php' );
```

Comment:
Instead of returning to `/wp-admin/themes.php`, shouldn't we return to either `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store` or `/wp-admin/admin.php?page=wc-admin`?

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/61991/commits/ac131bfbd64f86ea949dc9816ccd813b0886d143